gym-report-server/src/test/java/com/polot/gym/service/TrainingReportServiceImplTest.java

```diff
@@ -30,24 +33,93 @@ class TrainingReportServiceImplTest {
+                .trainingDuration(trainingDuration)
+                .type(type)
+                .build();
+
+        customTrainingReport = new CustomTrainingReport() {
+            @Override
+            public String getTrainerUsername() {
+                return trainerUsername;
+            }
+
+            @Override
+            public String getTrainerFirstname() {
+                return firstname;
+            }
+
+            @Override
+            public String getTrainerLastname() {
+                return lastname;
+            }
+
+            @Override
+            public Boolean getIsActive() {
+                return isActive;
+            }
+
+            @Override
+            public Integer getYear() {
+                return year;
+            }
+
+            @Override
+            public String getMonth() {
+                return "JANUARY";
+            }
+
+            @Override
+            public Integer getDurationSummary() {
+                return trainingDuration;
+            }
+        };
     }
 
     @Test
     void postReport() {
-        ReportRequest reportRequest = new ReportRequest("", "", "", true, LocalDate.now().toString(), 1, ReportType.ADD);
-        TrainingReport report = new TrainingReport();
-        given(trainingReportRepository.save(any())).willReturn(report);
-        trainingReportService.postReport(reportRequest);
-        verify(trainingReportRepository).save(any());
+        ReportRequest request = new ReportRequest(trainerUsername, firstname, lastname, isActive, LocalDate.of(year, java.time.Month.JANUARY, day).toString(), trainingDuration, type);
+        given(trainingReportRepository.save(trainingReport)).willReturn(trainingReport);
+        trainingReportService.postReport(request);
+        verify(trainingReportRepository,times(1)).save(TrainingReport.builder()
+                .trainerUsername(trainerUsername)
+                .trainerFirstname(request.getTrainerFirstname())
+                .trainerLastname(request.getTrainerLastname())
+                .isActive(request.getIsActive())
+                .year(LocalDate.parse(request.getTrainingDate()).getYear())
+                .month(Month.valueOf(LocalDate.parse(request.getTrainingDate()).getMonth().name()))
+                .date(LocalDate.parse(request.getTrainingDate()).getDayOfMonth())
+                .trainingDuration(request.getTrainingDuration())
+                .type(request.getType())
+                .build());
     }
 
     @Test
     void getAll() {
-        given(trainingReportRepository.findAllByGroupBy()).willReturn(List.of());
+        List<CustomTrainingReport> reports = Arrays.asList(customTrainingReport);
+        given(trainingReportRepository.findAllCustomTrainingReportsSummary()).willReturn(reports);
         List<ReportResponse> reportResponses = trainingReportService.getAll();
-        assertThat(reportResponses).isNotNull();
+        assertThat(reportResponses.size()).isEqualTo(1);
```

Comment:
we can verify not only the size of the list here, but also add some tests that verify our mappings and other logic.